### PR TITLE
Add logic to the url-analyzer job

### PR DESCRIPTION
**Pull request by jeroenpf**
This PR is part of a series (#53) of PRs and is preceded by #55.

Logic is added to extract colors from a website. It does so by trying to get background and link colors from the target URL and extracting dominant colors from the logo and favicon. This logic is ported from an earlier proof of concept we did.

## Test instructions

Firstly, it is important that the version of Node on the host machine matches the version of Node inside the container. Currently, the version is `14.19.1` - if these versions don't match, `npm install` will result in broken packages if ran from the host machine.  If matching the versions isn't possible, running `npm clean-install` inside the container will also work.

Run `npm install` to install new dependencies.

To test, simply hit the url-analyzer external endpoint with a URL, for example http://localhost:8000/url-analyzer?url=https://example.com

### CODE REVIEW

**Comment by roo2** [comment]
I left some comments on https://github.com/Automattic/mShots/pull/56 which apply to this PR, it looks like they implement a lot of the same code

**Comment by roo2** [comment]
node-canvas binaries aren't available for M1 macs', I tried installing them but then got issues where the application wouldn't start because it had been compiled for the wrong architecture. I tried running `npm install --target_arch=arm64 --target_platform=linux` etc and a few other things but haven't been able to run this from my machine.

**Comment by vishnugopal** [comment]
Think I'm running into the same issue as @roo2, when I try to run this, the worker restarts continuously:

```
[2022-04-08T10:14:27.248] [DEBUG] flog - The worker thread #5655 (pid:45410) is online.
[2022-04-08T10:14:27.260] [DEBUG] flog - worker thread #5655 (pid:45410) has disconnected.
[2022-04-08T10:14:27.260] [DEBUG] flog - worker thread #5655 (pid:45410) died (1). restarting.
[2022-04-08T10:14:27.260] [DEBUG] flog - sending "false" to forced reload parameter
[2022-04-08T10:14:27.262] [DEBUG] flog - The worker thread #5656 (pid:45417) is online.
```

and the result is always pending from the API:

```
{"http_status":200,"data":{"url":"https:\/\/example.com","status":"pending"}}
```

**Comment by vishnugopal** [comment]
Think I'm running into the same issue as @roo2, when I try to run this, the worker restarts continuously:

```
[2022-04-08T10:14:27.248] [DEBUG] flog - The worker thread #5655 (pid:45410) is online.
[2022-04-08T10:14:27.260] [DEBUG] flog - worker thread #5655 (pid:45410) has disconnected.
[2022-04-08T10:14:27.260] [DEBUG] flog - worker thread #5655 (pid:45410) died (1). restarting.
[2022-04-08T10:14:27.260] [DEBUG] flog - sending "false" to forced reload parameter
[2022-04-08T10:14:27.262] [DEBUG] flog - The worker thread #5656 (pid:45417) is online.
```

and the result is always pending from the API:

```
{"http_status":200,"data":{"url":"https:\/\/example.com","status":"pending"}}
```

**Comment by roo2** [comment]
@vishnugopal I was able to get the error message by adding the `-C` flag to the `start-stop-daemon`  command in 
[mshots_ctl.sh:72](https://github.com/Automattic/mshots/blob/master/mshots_ctl.sh#L72)
```bash
start-stop-daemon -S -C -q -m -b --pidfile /var/run/mshots.pid --exec /usr/local/node/bin/node --chdir $INSTALL_DIR -- $ARGS
```
Then opening a shell in the docker container and running:

./mshots_ctl.sh stop
./mshots_ctl.sh start

You should then see the error, in my case: 

```
Error: /opt/mshots/node_modules/canvas/build/Release/canvas.node: invalid ELF header
```

The _ELF header_ thing would be because my canvas.node binary was built for the wrong CPU architecture. I haven't figured out how to make it build for the right architecture yet!

**Comment by jeroenpf** [comment]
> 

I got the invalid ELF header error when i ran `npm install` on my host machine with a node version that did not match the version inside the container. Are you sure that the versions match so we can be certain it is not caused by this?

File: README.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ Docker can be used to run the mshots service on OSX for dev/testing purposes:
 
 - First, give docker permission to mount your dev directory. For Docker Desktop this is in
   `Preferences -> Resources-> File Sharing` where you can add the path to where mshots is checked out.
+- Make sure the version of Node on your system is the same as the one installed inside the docker container. This is version 14.19.1. If the versions do not match, the application could break.
```
[the remark this answers is not in the document]
Comment (jeroenpf) [reply README.md:66]: I've changed the version in `package.json`.  To my knowledge, there is no way to enforce this just by having it in the `package.json` file, it needs to be accompanied by a `engine-strict=true` configuration in `.npmrc`. Still, setting the right version in the `package.json` file is good so I've changed it.

**Comment by roo2** [comment]
@jeroenpf I think I tried that, installing the brew dependencies with `arch -arm64` and then tried playing around with the npm install command's --target-arch param. I couldn't get past that error with the ELF headers, I probably did something wrong or didn't do a clean install when I should have https://stackoverflow.com/questions/24961623/installing-node-js-packages-for-different-architecture

**Pull request merged**